x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TumblingWindow.java

```diff
@@ -47,8 +49,23 @@
  */
 public class TumblingWindow implements Executable {
 
+    private static final int CACHE_MAX_SIZE = 64;
+
     private final Logger log = LogManager.getLogger(TumblingWindow.class);
 
+    /**
+     * Simple cache for removing duplicate strings (such as index name or common keys).
+     * Designed to be low-effort, non-concurrent (not needed) and thus optimistic in nature.
+     * Thus it has a small, upper limit so that it doesn't require any cleaning up.
+     */
+    // start with the default size and allow growth until the max size
+    private final Map<String, String> stringCache = new LinkedHashMap<>(16, 0.75f, true) {
```

Comment:
One things to consider: Start with the CACHE_MAX_SIZE instead of the default size (`16`). The HashMap will have to grow anyways, unless you think that the chances of having <= 32 different strings are high.